apps/src/components/map-layers/map-legend-inner-s2d.tsx

```diff
@@ -249,27 +198,29 @@ export const MapLegendInnerS2D = () => {
 	};
 
 	const probabilityStatement: ProbabilityStatementProps = {
-		variableName: S2D_HARDCODED_LEGEND_VAR_NAME,
-		outcome: S2D_HARDCODED_LEGEND_OUTCOME,
+		variableName: 'total precipitation',
+		outcome: 'unusually high or low',
 		locale: lang,
 	};
 
 	let errorMessage: string | undefined
-	const scaleLength = data.scale.length;
-	const rowsLengthMustBe = scaleLength - 1;
-	for (const [idx, row] of (data?.rows ?? []).entries()) {
-		const { colors } = row;
-		const curLen = colors.length;
-		if (curLen !== rowsLengthMustBe) {
-			const message = `
-				Inconsistency error at row index ${idx}:
-				We're expecting to have exactly ${rowsLengthMustBe}
-				and we got an array of ${curLen} color codes
-			`
-				.replace(/(\n|\s){2,}/g, ' ')
-				.trim();
-			console.error(message);
-			errorMessage = message;
+	const scaleLength = (data?.scale ?? []).length;
+	if (scaleLength > 0) {
+		const rowsLengthMustBe = scaleLength - 1;
+		for (const [idx, row] of (data?.rows ?? []).entries()) {
```

Comment:
Same comment about `data` and its `rows` field being defined:

```suggestion
		for (const [idx, row] of data.rows.entries()) {
```